### PR TITLE
Clean up perl packages

### PR DESCRIPTION
**Pull request by druvus**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/perl-test-base/meta.yaml [hunk 1]
```diff
@@ -8,25 +8,23 @@ source:
...
-    - perl-threaded
+    - perl
     - perl-spiffy
     - perl-text-diff
 
 test:
   imports:
     - Test::Base::Filter
-  # Perl 'use' tests trigger "readline() on unopened filehandle DATA" errors,
-  # so we have to use this "dummy" test package instead.
```
Comment (mbargull) [inline recipes/perl-test-base/meta.yaml]: I suppose this is a useful comment as it explains the reason for the "non-standard" test case. I'd vote to keep it as long as that workaround is needed.

File: recipes/perl-test-yaml/meta.yaml [hunk 2]
```diff
@@ -8,20 +8,18 @@ source:
...
     - perl-test-base
 
   run:
-    - perl-threaded
+    - perl
     - perl-test-base
 
 test:
-  # Perl 'use' tests trigger "readline() on unopened filehandle DATA" errors,
-  # so we have to use this "dummy" test package instead.
```
Comment (mbargull) [inline recipes/perl-test-yaml/meta.yaml]: I suppose this is a useful comment as it explains the reason for the "non-standard" test case. I'd vote to keep it as long as that workaround is needed.

**Comment by mbargull** [comment]
Build failed for `perl-test-base` and `perl-test-yaml` (skipped) due to the single quotes in the test commands. This is a known issue with the usage of `mulled`. Until we fixed that, we'll have to use double quotes:
`- perl -e "use Test::Base tests => 1; pass;"`
`- perl -e "use Test::YAML tests => 1; pass;"`

Not sure why `perl-tie-ixhash` failed as the error message isn't very informative..

File: recipes/perl-test-base/meta.yaml (hunk 1 above)
Comment (druvus) [reply recipes/perl-test-base/meta.yaml]: good catch! I am adding it back.

**Pull request merged**